### ISSUE TITLE
Documentation for old versions

### ISSUE THREAD

**Issue by avilaton** [marshmallow-code/webargs#511]
Hi all,
I was looking at the documentation and was unable to find the docs for an outdated version I have to work with. The project I work on is using 1.4.0, crazy old, but the docs only show stable and dev docs. Would it be possible to keep the docs for legacy versions up there? 
If I'm pointed to the right place I can code up whatever needs to be done so that readthedocs builds old versions, I just don't know where that would be.
Thanks in advance for all the effort.

**Comment by sloria** [comment]
I'm hesitant to keep around old docs versions (esp. as old as 1.x), since I think that would require having another git branch lying around.

My suggestion is to just build the docs locally for the version you're using.

```sh
git clone https://github.com/marshmallow-code/webargs.git
cd webargs
git checkout 1.4.0
pip install -r docs/requirements.txt
inv docs -b
```

**Comment by avilaton** [comment]
Thanks for your reply.
I read from https://docs.readthedocs.io/en/stable/versions.html#tags-and-branches that tags are also a way to have docs for previous versions, no need to keep a version around. Marshmallow, to name one example, has had to keep version 2 around due to the importance of it being available.

Not having the docs around means less hits on google, the inability to make reference in stackoverflow and an extra complication to anyone wanting to update or figure out how something used to work. I like this project and version 1.4 was already so good to us that we moved to it a while back. We will try to update, but I beg you reconsider if this isn't a big complication.

**Comment by sloria** [comment]
It's not as turn-key as I hoped. I just tried to enable the docs builds for the latest tags on each of the unsupported release lines but the builds fail.

<img width="808" alt="Image 2020-05-06 at 5 58 27 PM" src="https://user-images.githubusercontent.com/2379650/81232611-33e10100-8fc3-11ea-847a-dd08ae71b53e.png">

I don't really have time to deal with the bit rot on our old docs versions--sorry I can't be more help.

**Comment by avilaton** [comment]
@sloria thanks in any case for giving it a shot. I appreciate you looking into that in any case, I thought I could be of help. If I hit the issue and figure out a solution, I'll try sending a PR.

**Comment by avilaton** [comment]
I made minor changes to a branch created from 1.4.0 to get readthedocs to build it and got it running. I made a tiny PR against a branch I made on my fork based on 1.4.0.
https://github.com/avilaton/webargs/pull/1/files
I think it would be a good addition to have docs for previous versions available. I also think I could make changes to sphinx on the 1.4.0 branch so that the theme is the same as the latest. That would make it consistent with the current theme and avoid. 

What do you think? I honestly regard postgresql docs as a reference case here, in which all previous version documentation pages are available.

Here is the built version
https://webargs-fork.readthedocs.io/en/1.4.0-docs/advanced.html

**Comment by sloria** [comment]
Is there a way to make it work without having to create a bunch of branches for each of the old release lines? 🤔  Also, won't the fix need to be forward-ported to every newer version. I'm not opposed to getting the fix in iff it's low effort and low-maintenance. I just don't want to have to maintain a bunch of old versions.

**Comment by avilaton** [comment]
I completely agree with you on the maintenance burden. On my fork, a tag is sufficient. In your repo, you would only have to keep a tag, I used 1.4.0-docs because nobody should touch the 1.4.0 tag. The docs show up and the branch can be deleted.
To be able to add it to your repo, I need to be able to send a PR to a branch created from 1.4.0. After the fix is merged and tagged, the branch can be deleted.

To minimize the maintenance burden, we could have for each tag the exact same sphinx theme and tooling. This also fixes the dev ux when changing from one version to the other.

In short, if you make a branch for me with `git checkout -b 1.4.0-docs 1.4.0` then I can submit a PR to it with the fix. Then after merging, you only need to add a tag and enable docs for that tag on readthedocs. 
I can take care of fixing docs for every version that is requested, maybe I do all the latest versions of mayors so we cover some ground.

**Comment by sloria** [comment]
> if you make a branch for me with git checkout -b 1.4.0-docs 1.4.0

Done.

> maybe I do all the latest versions of mayors so we cover some ground.

Sounds like a good plan. I'll make branches for those when I have some time.

**Comment by sloria** [comment]
OK, each release line now has a docs build. Thanks for taking care of this for 1.x @avilaton

**Comment by avilaton** [comment]
Thanks to you @sloria !
